autoload/neomake/makers/cargo.vim

```diff
@@ -1,11 +1,16 @@
 " vim: ts=4 sw=4 et
 
-function! neomake#makers#cargo#cargo()
+function! neomake#makers#cargo#cargo() abort
     return {
         \ 'args': ['build'],
         \ 'errorformat':
-            \   '%-Z%f:%l,' .
-            \   '%+C %s,' .
-            \   '%A%f:%l:%c: %*\d:%*\d\ %t%*[^:]: %m,',
+            \ '%Eerror[E%n]: %m,'.
+            \ '%Eerror: %m,'.
+            \ '%Wwarning: %m,'.
+            \ '%Inote: %m,'.
+            \ '%-Z\ %#-->\ %f:%l:%c,'.
```

Comment:
Ah ok, this particular -Z is actually the end of multiline message in vim's errorfmt (refer to [docs](http://vimdoc.sourceforge.net/htmldoc/quickfix.html#errorformat) to see more. Anyway thanks for pointing that out, i removed it from rust.vim.